platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c

```diff
@@ -192,27 +140,106 @@ hrt_absolute_time(void)
 
 #define REG(_reg)	_REG(HRT_TIMER_BASE + (_reg))
 
-#define rSC         REG(KINETIS_TPM_SC_OFFSET)
-#define rCNT        REG(KINETIS_TPM_CNT_OFFSET)
-#define rMOD        REG(KINETIS_TPM_MOD_OFFSET)
-#define rC0SC       REG(KINETIS_TPM_C0SC_OFFSET)
-#define rC0V        REG(KINETIS_TPM_C0V_OFFSET)
-#define rC1SC       REG(KINETIS_TPM_C1SC_OFFSET)
-#define rC1V        REG(KINETIS_TPM_C1V_OFFSET)
-#define rSTATUS     REG(KINETIS_TPM_STATUS_OFFSET)
-#define rCOMBINE    REG(KINETIS_TPM_COMBINE_OFFSET)
-#define rPOL        REG(KINETIS_TPM_POL_OFFSET)
-#define rFILTER     REG(KINETIS_TPM_FILTER_OFFSET)
-#define rQDCTRL     REG(KINETIS_TPM_QDCTRL_OFFSET)
-#define rCONF       REG(KINETIS_TPM_CONF_OFFSET)
+#define rSC         REG(0x0)
+#define rCNT        REG(0x4)
+#define rMOD        REG(0x8)
+#define rC0SC       REG(0xC)
+#define rC0V        REG(0x10)
+#define rC1SC       REG(0x14)
+#define rC1V        REG(0x18)
+#define rSTATUS     REG(0x50)
+#define rCOMBINE    REG(0x64)
+#define rPOL        REG(0x70)
+#define rFILTER     REG(0x78)
+#define rQDCTRL     REG(0x80)
+#define rCONF       REG(0x84)
+
+/* SC Bit Fields */
```

Comment:
Can we upstream the header files to nuttx and bring them in by inclusion?

Reply:
Yeah this was a work around for the header files, but I can make a FTM register header file and upstream to NuttX.

Reply:
OK if you do that I can backport it ASAP

Reply:
Done see https://github.com/apache/incubator-nuttx/pull/614

Reply:
Done https://github.com/PX4/NuttX/pull/95 - @dagar Please appove/merge